sdks/python/apache_beam/io/lyft/s3_and_kinesis.py

```diff
@@ -2,32 +2,76 @@
 import logging
 import random
 import string
-from collections import namedtuple
 
 from apache_beam import PTransform
 from apache_beam.pvalue import PBegin
 from apache_beam.pvalue import PCollection
 from apache_beam.transforms.core import Windowing
 from apache_beam.transforms.window import GlobalWindows
 
-Event = namedtuple('Event', 'name lateness_in_sec lookback_days')
-S3Config = namedtuple('S3Config', 'parallelism lookback_hours')
+
+class EventConfig(object):
+    """
+    Configuration of analytic event.
+    """
+    DEFAULT_MAX_OUT_OF_ORDERNESS_MILLIS = 5_000  # maximum allowed delay in time before an element is ignored
+
+    def __init__(self, name):
+        self.name = name                          # name of the analytics event.
+        self.max_out_of_orderness_millis = EventConfig.DEFAULT_MAX_OUT_OF_ORDERNESS_MILLIS
+        self.lookback_days = None                 # historical start time to consume events from.
+
+    def with_max_out_of_orderness_millis(self, out_of_orderness_millis):
+        """
+        The interval between the maximum timestamp seen so far and the watermark that
+        is emitted. For example, if this is set to 1000ms, after seeing a record for
+        10:00:01 we will emit a watermark for 10:00:00, indicating that we believe that all
+        data from before that time has arrived.
+        """
+        self.max_out_of_orderness_millis = out_of_orderness_millis
+        return self
+
+    def with_lookback_in_days(self, lookback_days):
+        self.lookback_days = lookback_days
+        return self
+
+
+class S3Config(object):
+    """
+    S3 configuration.
+    """
+    DEFAULT_S3_PARALLELISM = 1     # parallelism for s3 source connector. Defaults to 1.
+    DEFAULT_LOOKBACK_THREAD = 23   # threshold in hours for consuming events from S3.
```

Comment:
nit: lets change to `DEFAULT_LOOKBACK_THRESHOLD_HOURS`

Reply:
Good catch! I had a typo.